cvat/apps/engine/serializers.py

```diff
@@ -3499,61 +3771,74 @@ class Meta:
-        storage_instance = getattr(instance, storage_type)
+                    similar_ones.append(cs)
 
-        if new_cloud_storage_id:
-            if new_location and new_location != models.Location.CLOUD_STORAGE:
-                raise serializers.ValidationError(
-                    f"It is not allowed to specify '{new_location}' location together with cloud storage id"
-                )
-            elif (
-                not new_location
-                and getattr(storage_instance, "location", None) != models.Location.CLOUD_STORAGE
-            ):
-                raise serializers.ValidationError(
-                    f"The configuration of {storage_type} is not full"
-                )
+                if len(similar_ones) != 1:
+                    raise serializers.ValidationError(msg_no_similar_cs)
 
-            if not models.CloudStorage.objects.filter(id=new_cloud_storage_id).exists():
-                raise serializers.ValidationError(
-                    f"The specified cloud storage {new_cloud_storage_id} does not exist."
-                )
-        else:
-            if new_location == models.Location.CLOUD_STORAGE:
-                raise serializers.ValidationError(
-                    "Cloud storage was selected as location but its id was not specified"
-                )
-            elif (
-                not new_location
-                and getattr(storage_instance, "location", None) == models.Location.CLOUD_STORAGE
-                and "cloud_storage_id" in new_conf
-            ):
-                raise serializers.ValidationError(
-                    "It is not allowed to reset a cloud storage id without explicitly resetting a location"
-                )
+                new_conf = {
+                    "location": models.Location.CLOUD_STORAGE,
+                    "cloud_storage_id": similar_ones[0].pk
+                }
+            except models.CloudStorage.DoesNotExist:
+                # generally should not occur
+                new_conf = {
+                    "location": models.Location.LOCAL,
+                    "cloud_storage_id": None,
+                }
 
-        if not storage_instance:
-            storage_instance = models.Storage(**new_conf)
-            storage_instance.save()
-            setattr(instance, storage_type, storage_instance)
-            continue
+        storage_serializer = StorageSerializer(storage_instance, data=new_conf)
+        storage_serializer.is_valid(raise_exception=True)
+
+        # todo: check that CS exists (filter also by dst_organization_id or instance.organization_id)
 
-        storage_instance.location = new_location or storage_instance.location
-        storage_instance.cloud_storage_id = new_cloud_storage_id
-        storage_instance.save()
+        if storage_instance:
+            storage_serializer.save()
+        else:
+            storage_instance = storage_serializer.save()
+            setattr(instance, storage_type, storage_instance)
```

Comment:
```suggestion
        storage_instance = storage_serializer.save()
        setattr(instance, storage_type, storage_instance)
```
- Why this condition is needed? 
- The updated existing object might be outdated in the instance, but will be present if there was no object

Reply:
you're right